### PR TITLE
feature(convenience): item view zoom on scroll + ctrl

### PR DESCRIPTION
**Pull request by justintime4tea**
## Feature Concept/Behavior
When scroll events are in the "item_view" (in tab), such as a mouse scrolling the mouse wheel, and the CTRL modifier is held, zoom in and out (increase/decrease icon size). 

This finishes resolving [#18](https://github.com/pop-os/cosmic-files/issues/18) by adding scroll zoom. Zoom via Ctrl+PLUS and Ctrl+MINUS were already implemented.

### "Zooming" grid_view

screenshot-01

### "Zooming" list_view

listshot

## Implementation
I used the existing "zoom mechanism" provided by "tab", ZoomIn/ZoomOut. (EDIT) ~~, and simply added two more Message, ScrollUp and ScrollDown.~~

The MouseArea now allows you to capture the scroll of the wheel (on_scroll) including any active modifiers (CTRL, SHIFT, etc...). Within "tab", on_scroll is used to produce ZoomIn and ZoomOut messages in order to drive the "zoom on ctrl+scroll" behavior.

## Extra notes
I added tests. I'm not sure if they were expected or expansive enough, or good enough, but I thought, what the heck.

None of this code is my "baby". Please shred it apart and tell me how it can be made better or stand a better chance at inclusion in the codebase.

### CODE REVIEW

**Comment by l-const** [comment]
Hey @justintime4tea  nice work , this should close https://github.com/pop-os/cosmic-files/issues/18#issue-2083994408 #18 now that scroll zoom is implemented, so you should probably link it. Again,  thanks for your work.

File: src/scroll_area.rs [hunk 1]
```diff
@@ -0,0 +1,220 @@
+//! A container for reacting to scroll events.
```
[the remark this answers is not in the document]
Comment (justintime4tea) [reply src/scroll_area.rs]: Thanks for the feedback, Jack! I appreciate the opportunity to dive back in and make things a bit better. Let me know your thoughts on the latest changes! Appreciate you looking at this.

**Comment by justintime4tea** [comment]
> Hey @justintime4tea nice work , this should close [#18 (comment)](https://github.com/pop-os/cosmic-files/issues/18#issue-2083994408) #18 now that scroll zoom is implemented, so you should probably link it. Again, thanks for your work.

Thanks for the warm welcome! I'll get the issue linked, that's a splendid idea.

**Comment by l-const** [comment]
> > Hey @justintime4tea nice work , this should close [#18 (comment)](https://github.com/pop-os/cosmic-files/issues/18#issue-2083994408) #18 now that scroll zoom is implemented, so you should probably link it. Again, thanks for your work.
> 
> Thanks for the warm welcome! ~I'll get the issue linked, that's a splendid idea.~ (EDIT) How do I go about linking the issue? Or do you mean, mentioning it with a link like you have, or in the description, so that it shows up in the issue's activity? Appreciate the help.

Here is the github docs for doing it: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword , essentially either in a commit or in the description of the PR. Still not that important just remember to close the issue.

**Review by jackpot51 — APPROVED** [review]

**Pull request merged**